### PR TITLE
Created Video listing page.Kindly review it and let me know what do you think about it !

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**

### CODE REVIEW

File: src/Pages/VideoListingPage/VideoListingPage.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { VideoCard } from '../../Shared';
+import './VideoListingPage.css';
+import { getCategoryById , getVideos, getAllCategory} from '../../Helpers/Services/actions';
+import { v4 as uuid } from "uuid";
+
+const VideoListingPage = () =>{
+
+    const [data, setData] = useState({});
+    const [ videoData, setVideoData] = useState([]);
+    const [ categoryData, setCategoryData] = useState([]);
```
Comment (Sahil25061999) [inline src/Pages/VideoListingPage/VideoListingPage.jsx:13]: try using useReducer if useState is being used more than 2 times

File: src/Pages/VideoListingPage/VideoListingPage.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            })
+        }
+    ,[])
+
+    useEffect(()=> {
+        getAllCategory()
+        .then(res=> {
+            if(res.status === 200){
+                setCategoryData(res.data.categories);
+                const sel_ctgry = res.data.categories.find(ctgry => ctgry._id === categoryId);
```
Comment (Sahil25061999) [inline src/Pages/VideoListingPage/VideoListingPage.jsx:53]: This probably can be written in the same useEffect, not sure though.

File: src/Pages/VideoListingPage/VideoListingPage.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                <div className="chip-word-flex">
+                {data && Object.keys(data).length?
+                    data.subcategory.length?
+                        data.subcategory.map((sub_ctgry,cIndx)=> 
+                            <p key={cIndx} className="chip-word">{sub_ctgry}</p>
+                        )
+                    :<p key={cIndx} className="chip-word">All</p>
+                :<p className="chip-word">All</p>}
+                </div>
+            </div> */}
```
Comment (Sahil25061999) [inline src/Pages/VideoListingPage/VideoListingPage.jsx:78]: remove the commented code if not necessary.

**Review by Sahil25061999 — APPROVED** [review]
Looks good to me, just require minor changes, a UI suggestion you can try different color for the side navbar

File: src/Pages/VideoListingPage/VideoListingPage.jsx (hunk 2 above)
Comment (kgajriaAgarwal) [reply src/Pages/VideoListingPage/VideoListingPage.jsx:53]: this is done to increase the code redability

File: src/Pages/VideoListingPage/VideoListingPage.jsx (hunk 3 above)
Comment (kgajriaAgarwal) [reply src/Pages/VideoListingPage/VideoListingPage.jsx:78]: this is for future perspective

File: src/Pages/VideoListingPage/VideoListingPage.jsx (hunk 1 above)
Comment (kgajriaAgarwal) [reply src/Pages/VideoListingPage/VideoListingPage.jsx:13]: ok!

**Review by sri-rishi — APPROVED** [review]
code looks good, nice work , feel free to merge

**Pull request merged**